.github/workflows/test-tpg.yml

```diff
@@ -14,36 +20,89 @@ on:
+        else
+          echo "has_changes=true" >> $GITHUB_OUTPUT
+        fi
+    - name: Get Job URL
+      if: ${{ !failure() && steps.pull_request.outputs.has_changes == 'true' }}
+      id: get_job
+      run: |
+        response=$(curl --get -Ss -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" -H "Accept: application/vnd.github.v3+json" "https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/attempts/${{ github.run_attempt }}/jobs")
+        html_url=$(echo "$response" | jq -r --arg job_name "${{ github.job }}" '.jobs | map(select(.name == $job_name)) | .[0].html_url')
+        echo "url=${html_url}" >> $GITHUB_OUTPUT
+    - name: Post Pending Status to Pull Request
+      if: ${{ !failure() && steps.pull_request.outputs.has_changes == 'true' }}
+      run: |
+        curl -X POST -H "Authorization: token ${{secrets.GITHUB_TOKEN}}" \
+        -H "Accept: application/vnd.github.v3+json" \
+        "https://api.github.com/repos/GoogleCloudPlatform/magic-modules/statuses/${{github.event.inputs.sha}}" \
+        -d '{
+        "context": "${{ github.event.inputs.repo }}${{ env.status_suffix }}",
+        "target_url": "${{ steps.get_job.outputs.url }}",
+        "state": "pending"
+        }'
     - name: Set up Go
-      uses: golang/setup-go@v2
+      if: ${{ !failure() && steps.pull_request.outputs.has_changes == 'true' }}
+      uses: actions/setup-go@v4
       with:
-        go-version: '1.19.x'
-    - name: Build
+        go-version: '^1.19.1'
+    - name: Build Provider
+      if: ${{ !failure() && steps.pull_request.outputs.has_changes == 'true' }}
       run: |
         cd provider
         go build
-    - name: Lint
+    - name: Run Unit Tests
+      if: ${{ !failure() && steps.pull_request.outputs.has_changes == 'true' }}
+      run: |
+        cd provider
+        make test
+    - name: Lint Check
+      if: ${{ !cancelled() && steps.pull_request.outputs.has_changes == 'true' }}
       run: |
         cd provider
         make lint
-      if: ${{ always() }} # This step will always run, even if the previous fails
-    - name: Test
+    - name: Documentation Check
+      if: ${{ !cancelled() && steps.pull_request.outputs.has_changes == 'true' }}
       run: |
         cd provider
-        make test
-      if: ${{ always() }} # This step will always run, even if the previous fails
-
-
+        make docscheck
+    - name: Post Result Status to Pull Request
+      if: ${{ !cancelled() && steps.pull_request.outputs.has_changes == 'true' }}
+      run: |
+        curl -X POST -H "Authorization: token ${{secrets.GITHUB_TOKEN}}" \
+        -H "Accept: application/vnd.github.v3+json" \
+        "https://api.github.com/repos/GoogleCloudPlatform/magic-modules/statuses/${{github.event.inputs.sha}}" \
+        -d '{
+        "context": "${{ github.event.inputs.repo }}${{ env.status_suffix }}",
+        "target_url": "${{ steps.get_job.outputs.url }}",
+        "state": "${{ job.status }}"
```

Comment:
the `job.status` can be `success`, `failure`, or `cancelled`, but the `status.state` field can `error`, `failure`, `pending`, `success`.

If the job is cancelled this would fail - but we're not running this step if the job is cancelled, so it doesn't matter. 👍